File: UI/Hu/src/app/events/models/team.model.ts
import { Gender } from "../enums/gender.enum";
import { Country } from "./country-model";
import { FieldTranslations } from "./field-translation.model";
import { Sport } from "./sport-model";
import { TeamColors } from "./team-colors.model";

export interface Team {
    name:               string;
    slug:               string;
    shortName:          string;
    gender?:            Gender;
    sport:              Sport;
    userCount:          number;
    nameCode:           string;
    disabled?:          boolean;
    national:           boolean;
    type:               number;
    id:                 number;
    country:            Country;
    subTeams:           any[];
    teamColors:         TeamColors;
    fieldTranslations?: FieldTranslations;
}

File: API/Hu.API/Hu.API/Models/TeamModel.cs
using Newtonsoft.Json;

namespace Hu.API.Models
{
    public class TeamModel
    {
        [JsonProperty("name")]
        public string? Name { get; set; }

        [JsonProperty("slug")]
        public string? Slug { get; set; }

        [JsonProperty("shortName")]
        public string? ShortName { get; set; }

        [JsonProperty("gender")]
        public string? Gender { get; set; }

        [JsonProperty("sport")]
        public SportModel? Sport { get; set; }

        [JsonProperty("userCount")]
        public int UserCount { get; set; }

        [JsonProperty("nameCode")]
        public string? NameCode { get; set; }

        [JsonProperty("disabled")]
        public bool Disabled { get; set; }

        [JsonProperty("national")]
        public bool National { get; set; }

        [JsonProperty("type")]
        public int Type { get; set; }

        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("country")]
        public CountryModel? Country { get; set; }

        [JsonProperty("subTeams")]
        public List<object>? SubTeams { get; set; }

        [JsonProperty("teamColors")]
        public TeamColorsModel? TeamColors { get; set; }

        [JsonProperty("fieldTranslations")]
        public FieldTranslationsModel? FieldTranslations { get; set; }
    }
}
